src/middlewared/middlewared/alert/source/disk_temp.py

```diff
@@ -0,0 +1,53 @@
+# Copyright (c) - iXsystems Inc. dba TrueNAS
+#
+# Licensed under the terms of the TrueNAS Enterprise License Agreement
+# See the file LICENSE.IX for complete terms and conditions
+
+from middlewared.alert.base import (
+    Alert,
+    AlertCategory,
+    AlertClass,
+    AlertLevel,
+    ThreadedAlertSource,
+)
+
+
+class DiskTemperatureTooHotAlertClass(AlertClass):
+    category = AlertCategory.HARDWARE
+    level = AlertLevel.CRITICAL
+    title = "Disk Temperature Is Too Hot"
+    text = (
+        "Disk %(n)s (with serial: %(s)s) critical temperature"
+        " threshold is %(t)d degrees celsius and the current temp is"
+        " %(ct)d degrees celsius"
+    )
+
+
+class DiskTemperatureTooHotAlertSource(ThreadedAlertSource):
```

Comment:
No need to make this sync, no blocking I/O or CPU-intense operations are performed

Reply:
good point, fixed it.